smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/ProcessorMediator.java

```diff
@@ -234,44 +234,92 @@ private void processMethodReturningAPublisherOfPayloadsAndConsumingPayloads() {
     private void processMethodReturningIndividualMessageAndConsumingIndividualItem() {
         // Item can be message or payload
         if (configuration.consumption() == MediatorConfiguration.Consumption.PAYLOAD) {
-            this.processor = ReactiveStreams.<Message> builder()
-                    .flatMapCompletionStage(managePreProcessingAck())
-                    .map(input -> (Message) invoke(input.getPayload()))
-                    .buildRs();
+            if (configuration.isBlocking()) {
+                this.processor = ReactiveStreams.<Message> builder()
+                        .flatMapCompletionStage(managePreProcessingAck())
+                        .<Message> flatMapCompletionStage(input -> ((Uni<?>) invokeBlocking(input.getPayload()))
+                                .subscribeAsCompletionStage()
+                                .thenApply(result -> (Message) result))
+                        .buildRs();
```

Comment:
There is a bug here. It may have been there before.
POST_PROCESSING acknowledgment is not handled correctly (well not handled at all).

Reply:
I was wondering that yesterday, as it seemed it was the only one without it. Should've asked then!

Reply:
Can you open an issue?

Reply:
https://github.com/smallrye/smallrye-reactive-messaging/issues/480